Update revealjs v5 and dependencies

cderv opened the pull request:
Towards 
- #4795 


- Revealjs 5.1.0, new JS and themes
- Reveal Plugins 4.2.5 for chalkboard 2.3.3

This PR includes docs and process to update revealjs theming. It helps understand how this currently works to maps Sass variables from user's theming, to quarto internals, and then to expected Reveal.js own SASS theming system. 

We probably need more tests as most of the change will have effect in browser running JS code for reveal, so we don't have much test with this yet. 

## Manual testing

- [x] Demo slides: https://quarto.org/docs/presentations/revealjs/demo/ - Works OK ! 
- [x] All simpler demo examples from: https://quarto.org/docs/presentations/revealjs/demo/mini - Works OK ! 
- [x] All examples from https://quarto.org/docs/presentations/revealjs/examples - Works OK ! 
- [x] Checking all the themes change - I made a website to help with this : https://cderv.github.io/quarto-revealjs-themes/

cderv committed:
Update revealjs deps

- revealjs 5.0.5
- chalkboard 4.2.5

cderv committed:
Run updates to get all the new files

cderv committed:
Use our own copyTo version which will remove before copying

cderv committed:
Adapt current bootswatch themes patch function to use also on revealjs theme

cderv committed:
Add comment about why only two of four scss from revealjs are used

cderv committed:
Add functions to patch the themes to replace revealjs variable by Quarto ones

cderv committed:
incomplete patched themes

cderv committed:
improve test to avoid error

cderv committed:
Ignore white and black theme, that are aliased in quarto

they use default and dark and are adapted

cderv committed:
fix settings.scss patching

cderv committed:
Patch each theme to fully port them to quarto

cderv committed:
Add patch for input-panel-bg

cderv committed:
Add a patch for code block fixup

cderv committed:
Add a port of settings.scss even if not used directly

This is useful to check the variable inclusion

cderv committed:
Rename patch to puglin to match naming scheme

cderv committed:
Update dev doc for revealjs dependencies update

cderv committed:
CodeBackground is  not global to revealjs

cderv committed:
reveal update - Simpler patch file and insure patch is applied

otherwise error with leftover files
and ignore whitespace

cderv committed:
Reapply patch correctly for pdf-export.js

cderv committed:
update revealjs 5.1.0

cscheid commented:
This looks great!

Keeping in mind the upcoming posit::conf presentations, we should either:

- merge ASAP (as in today)
- or the day after posit::conf is done

What do you think?

cderv commented:
😄 definitely the main question. I ask myself the same, hence the draft state. We have a stable release working we could also consider that anyone doing a presentation for "production" (posit::conf) would use a stable version and not a pre-release 😉 Though our users likes to live on the edge of new stuff ! 😆 

I don't think the effect if big, but we don't really have test regarding how feature works in in a live browser situation so it requires manual testing. And I am just starting them. 

Could be small or no effect, but could also be the opposite 🤷‍♂️. So considering the milestone of conf, we should probably let this in a branch. This also leave me time to add to this, rebasing often on main and adding some tests. 

What is your take on this ?

cscheid commented:
You have the hands-on experience.

> So considering the milestone of conf, we should probably let this in a branch.

That works for me. Let's plan to merge right after conf.

cderv commented:
First issue to solve 
````markdown
---
title: "title"
format: 
  revealjs:
    chalkboard: true
---

## Quarto

Quarto enables you to weave together content and executable code into a finished document. To learn more about Quarto see <https://quarto.org>.

````

![image](https://github.com/user-attachments/assets/8ca89e6a-70b7-41bc-b59d-51c873adc363)

We get blurry screen by default following the update of chalkboard. 😭

cderv commented:
Nevermind, it seems it is fixed upstream: 
- https://github.com/rajgoel/reveal.js-plugins/pull/180

We need an unreleased version of chalkboard.

cderv committed:
Use a patch version of reveal-chalkboard plugin

this is an unreleased version with a fix for Reveal.js 5.0.3 change regarding overlay styling

cderv committed:
Updating beige theme changed the hash of the css file

cderv committed:
Add a changelog entry

[skip ci]

cderv commented:
Very small theme change after all, and it is now more inline with revealjs own theme. 

All good to merge this PR so I'll go ahead. 

I'll update documentation for 1.6 to close #4795 , including adding an highlight

cderv committed:
Add dracula among drak themes

cscheid commented on tests/docs/smoke-all/2024/05/03/9548.qmd:
Ok. That's unfortunate, but currently worth the additional pain of checking the test.

Let's keep an eye on this kind of thing, just so we don't let ourselves get bogged down with lots of manual updates on tests.
